codegen/src/test/resources/software/amazon/awssdk/codegen/poet/waiters/query-sync-waiter-class.java

```diff
@@ -66,6 +66,15 @@ private static String errorCode(Throwable error) {
         return postOperationSuccessWaiter.run(() -> client.aPostOperation(applyWaitersUserAgent(aPostOperationRequest)));
     }
 
+    @Override
+    public WaiterResponse<APostOperationResponse> waitUntilPostOperationSuccess(APostOperationRequest aPostOperationRequest,
+                                                                                WaiterOverrideConfiguration overrideConfig) {
+        Waiter<APostOperationResponse> waiter = Waiter.builder(APostOperationResponse.class)
+                                                      .acceptors(postOperationSuccessWaiterAcceptors())
+                                                      .overrideConfiguration(postOperationSuccessWaiterConfig(overrideConfig)).build();
+        return waiter.run(() -> client.aPostOperation(applyWaitersUserAgent(aPostOperationRequest)));
```

Comment:
Is it possible to just overload the Waiter `run()` method to take an the override instead? That way the existing waiter can be reused without instantiating a new one when a override is given. Would also simplify codegen

Reply:
Instantiating a new waiter is actually very cheap, but I do like your idea. Well, too bad I already made the codegen change :( Anyway, I'll update it

Reply:
But hopefully less code to maintain :). Also customers using the generic `Waiter` directly would benefit as well